### DISCUSSION TITLE
Pulse UI

### DISCUSSION THREAD

**Discussion by kennymark** [pulsecron/pulse#33] [Ideas]
Is there going to a compatible UI one could use together with Pulse just like agendash??

**Comment by code-xhyun** [comment]
> Is there going to a compatible UI one could use together with Pulse just like agendash??

Not yet! @kennymark 
When pulsecron, an ongoing cloud project, is completed, the Dashboard UI will be separated to create a Dashboard UI open source project.

**Reply by ronenteva** [reply]
@code-xhyun any estimation when that will happen?

**Reply by Romakita** [reply]
Hello @code-xhyun 
Do you need help to develop this feature ?

Ts.ED Framework provides a module to install pulse like is it with @tsed/agenda, but the dashboard feature is missing.
You can find our dedicated page here:
https://tsed.io/tutorials/pulse.html

And agenda here:
https://tsed.io/tutorials/agenda.html

It would be awesome to have the dashboad feature like is presented on your website <3

Also it could be cool to add Ts.ED on your website as framework using pulse.

Let me know if I can do something ;).

See you

**Reply by code-xhyun** [reply]
@Romakita Sure. I need help..

**Reply by Romakita** [reply]
There is code base to start the open source dashboard?

**Reply by code-xhyun** [reply]
@ronenteva  @Romakita I'm sorry for the delayed response. The schedule for the existing homepage work has been significantly postponed, so I'm considering starting an open-source dashboard project based on that code first. If I decide to do it, I'll let you know right away after creating the project.

**Comment by nullxx** [comment]
I've tested pulse with [agendash](https://github.com/agenda/agendash) and it works perfectly

**Reply by LorianeE** [reply]
Hi ! How did you do it? I tried to use Agendash with Pulse, but I encounter this error : 
```
 error_stack: 'TypeError: agenda.on is not a function\n' +
    '    at module.exports (.../node_modules/agendash/lib/controllers/agendash.js:8:10)\n' +
    '    at module.exports (.../node_modules/agendash/app.js:10:57)\n' +
```
It seems to happen when I use the "use" method like this : "app.use("/dash", Agendash(pulse));"

**Reply by snazzyj** [reply]
@LorianeE what version of Agendash are you using?  
 
doing the following works for Agendash @4.4.0
```
import agendash from 'agendash'
import { pulse } from './worker/pulse'
app.use('/pulse-jobs',  agendash(pulse))
```

**Reply by LorianeE** [reply]
Hi, sorry for the late reply. Actually it was a problem on my side : I'm using the awesome framework Ts.ED with the Agenda plugin and was trying to use the Pulse plugin. It was not 100% compatible because it wrapped only some methods from pulse, and "on()" was not one of them. I saw with the owner of Ts.ED and he did a fix, now everything works great (he doesn't use a wrapper anymore) and you're right, Agendash works with Pulse!
Thank you
